File: 118-pascals-triangle/118-pascals-triangle.py
class Solution:
    def generate(self, numRows: int) -> List[List[int]]:
        
        if numRows==1: return [[1]]
        
        res = [[1],[1,1]]
        
        for i in range(2,numRows):
            lis = []
            temp = res[-1]
            lis.append(temp[0])
            for i in range(1,len(temp)):
                lis.append(temp[i] + temp[i-1])
            lis.append(temp[-1])
            res.append(lis)
        
        return res

File: 118-pascals-triangle/118-pascals-triangle.cpp
class Solution {
public:
    vector<vector<int>> generate(int numRows) {
        vector<vector<int>> res {{1},{1,1}};
        if(numRows==1) return {res[0]};
        for(int i=2;i<numRows;i++){
            vector<int> temp = res[i-1];
            vector<int> lis;
            lis.push_back(temp[0]);
            for(int j=1;j<temp.size();j++) lis.push_back(temp[j]+temp[j-1]);
            lis.push_back(temp[temp.size()-1]);
            res.push_back(lis);
        }
        return res;
    }
};